assets/js/infinitescroll.js

```diff
@@ -34,7 +33,29 @@ $(function ($) {
         ticking = true;
     }
 
+    function sanitizePathname(path) {
+        var paginationRegex = /(?:page\/)(\d)(?:\/)$/i;
+
+        // remove hash params from path
+        path = path.replace(/#(.*)$/g, '').replace('////g', '/');
+
+        // remove pagination from the path and replace the current pages
+        // with the actual requested page. E. g. `/page/3/` indicates that
+        // the user actually requested page 3, so we should request page 4
+        // next, unless it's the last page already.
+        if (path.match(paginationRegex)) {
+            currentPage = parseInt(path.match(paginationRegex)[1]);
```

Comment:
Side-effects in single-purpose functions like this are best avoided so that it's easier to follow what's happening.

As an example you could rename this `getPaginationInfo(pathname)` then return an object...

```javascript
var currentPage = 1;

function getPaginationInfo(pathname) {
    ...
    return {
        currentPage: currentPage
        basePath: path
    }
}

function infiniteScroll() {
    var pagination = getPaginationInfo(window.location.pathname);

    // update the outer scope variable to keep track of which page we're on after consecutive loads
    currentPage = pagination.currentPage;

    ...

    var nextPage = pagination.basePath + 'page/' + currentPage + '/';

    ....
}
```

It's still not brilliant but it removes side-effects from the function and paves the way for a later refactor where you have an `InfiniteScroll` object and you can use `this.basePath` and `this.currentPage` rather than mixing global and local variables together.

Reply:
Missed it before but this is a good example of what I meant by wrapping the functionality in an object https://github.com/TryGhost/Casper/pull/438/files#diff-c1f39f81b633ffceb749e816cb50c816. With a few changes it would also allow you to pass in the `maxPages` variable so that we're not relying on globals, eg:

```
<script>
    var maxPages = parseInt('{{pagination.pages}}');
    InfiniteScroll.init({maxPages: maxPages});
</script>
```

Reply:
Awesome, thanks @kevinansfield! I tried to refactor it, but it took me then to long, so I split it and focused on fixing the bug first. I added this to my todo list for a clean up or bugfix day.